contracts/BoundaryManager.sol

```diff
@@ -91,8 +96,14 @@ contract BoundaryManager {
 
     function addRefundReceivers(address tokenAddress, address[] calldata refundReceivers) public {
         RefundBoundary storage safeRefundBoundary = safeRefundBoundaries[msg.sender][tokenAddress];
+        
+        // COMMENT: can this overflow the uint16 type and thus start the count from 0?
         safeRefundBoundary.allowedRefundReceiversCount = safeRefundBoundary.allowedRefundReceiversCount + uint16(refundReceivers.length);
 
+        // COMMENT: it would be good to add comment / intention of what's happening here. 
+        // Do I understand right that the code checks if the flag is already set to true (in which case it reverts). 
+        // Should it revert? Why not ignore?
```

Comment:
My goal was to prevent passing duplicate addresses and therefore messing up the counter. Same thing in the other function